www/docs/client/tanstack-react-query/usage.mdx

```diff
@@ -225,3 +225,31 @@ function SubscriptionExample() {
   return <>{/* ... */}</>;
 }
 ```
+
+### Inferring Input and Output types
+
+Available for all query procedures as well as routers and sub-routers. Allows creating [query filters](https://tanstack.com/query/latest/docs/framework/react/guides/filters#query-filters) in a type-safe manner.
+
+```ts
+import { AppRouter } from './path/to/server'
+
+//
+// Infer the input and output types of a full router
+
+import type { inferRouterInputs, inferRouterOutputs } from '@trpc/server';
+
+export type Inputs = inferRouterInputs<AppRouter>
+export type Outputs = inferRouterOutputs<AppRouter>
+
+//
+// Infer types for a single procedure
+
+import type { InferInput, InferOutput } from '@trpc/tanstack-react-query';
```

Comment:
Does this exist?

Reply:
Yes I added them to support the polymorphism work for this

Reply:
Why do I feel like this should either be 

- `inferInput`, or
- `InferredInput`

i can't motivate it but it feels wrong right now